libcperciva/datastruct/seqptrmap.c

```diff
@@ -74,7 +79,18 @@ seqptrmap_add(struct seqptrmap * M, void * ptr)
 void *
 seqptrmap_get(struct seqptrmap * M, int64_t i)
 {
-	size_t pos = i - M->offset;
+	size_t pos;
+	int64_t pos_i;
+
+	/* Sanity check, but we cannot fail, so don't use an assert. */
+	if (i < 0)
+		return (NULL);
```

Comment:
... and why checking for negative values anyway?  Why not just check if `i < M->offset` ?